blip-tap.md

```diff
@@ -924,21 +924,26 @@ accept if favorable
 
 ##### Request For Quote (`tap_rfq`)
 
-When a receiver wishes to receive `N` units of TAP asset ID `asset_id`, a new
+When a receiver wishes to receive `N` units of TAP asset ID `in_asset_id`, a new
 p2p message `tap_rfq` is sent with the following structure:
 
 1. type: ?? (`tap_rfq`)
 2. data:
      * [`32*byte`:`rfq_id`]
-     * [`32*byte`:`asset_id`]
+     * [`32*byte`:`in_asset_id`]
+     * [`32*byte`:`out_asset_id`]
```

Comment:
Seeing it laid out now, do we also need to split the `asset_amt` field as well? 

Before this change, `asset_amt` here was the amount of USD (or w/e) tied to the invoice. So we were asking for a price quote in BTC/`asset_id`. 

My interpretation now is that if you only specify `in_asset_id` (assume USD-X), then in the LN setting, it's a quote for _receiving_ `asset_id`. If you only specify `out_asset_id`, then in the LN setting, it's a quote for _sending_ `asset_id`. 

Does that match your interpretation? Perhaps we can make it a bit more concrete by outlining some typical user flows near the end of the document.

Reply:
In my opinion, the simplest `tap_rfq` scheme would be:
```
* [`32*byte`:`rfq_id`]

* [`32*byte`:`in_asset_id`]
* [`32*byte`:`out_asset_id`]

* [`BigSize`:`in_asset_amt`]
* [`BigSize`:`out_asset_amt`]
 ```
 
My reasoning:

 1. This allows us to quote for a specific amount of each asset. `suggested_rate_tick` does not specify volume. The "exchange rate" is likely to vary between transactions involving small and large volumes of assets.
 
 2. Two amount fields is simpler than `suggested_rate_tick`, the latter is a comparatively complicated field to explain and implement.
 
 3. We could always extend this schema in the future to support partial fills. We could add a flag field to indicate that the quote may be satisfied in part (but within the agreed rate). The state of that flag is, IMO, another factor to negotiate between peers, it shouldn't be assumed. Implementing as a separate flag facilitates that negotiation. But I don't think we should include that field right now as we're only matching payments to quotes on exact amounts 1:1.